lib/main.js

```diff
@@ -1,53 +1,72 @@
 'use strict'
 
+// like String.prototype.search but returns the last index
+function _searchLast (str, rgx) {
+  const matches = Array.from(str.matchAll(rgx))
+  return matches.length > 0 ? matches.slice(-1)[0].index : -1
+}
+
 function _interpolate (envValue, environment, config) {
```

Comment:
The function now works as follows:

1. Find the right-most group of variables (it can be a single `$VARIABLE` or a nested one (`${VARIABLE:-${NESTED:-default}}`);
2. Expand the inner-most variable;
3. Repeat from point 1 until no more variables are found;